web/api/v1/api.go

```diff
@@ -1351,6 +1352,25 @@ func (api *API) serveTSDBStatus(*http.Request) apiFuncResult {
 	}, nil, nil, nil}
 }
 
+type walReplayStatus struct {
```

Comment:
This struct is already defined once in head, can we reuse it? 🤗

Reply:
My issue here was the JSON mappings. They're required for the endpoint to work and I didn't know if we wanted them in structs not defined specifically for the API. Also, I wonder how the mutex would decode?. It might end up being private anyway, but I'm not sure.

Reply:
Also, now we have the `WALReplayState`, which should probably be returned by the API in its string format, which requires a different struct definition.

Reply:
We can add mapping TBH to anything, but I think it's common for our trust to reproduce those, so not a big deal